framework/include/interfaces/MortarInterface.h

```diff
@@ -0,0 +1,76 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "MooseTypes.h"
+
+#include <set>
+
+class InputParameters;
+class MooseObject;
+class FEProblemBase;
+class MooseMesh;
+class MortarData;
+
+/**
+ * An interface for accessing mortar mesh data
+ *
+ * \note moi is shorthand for mortar interface, so \p _moi_problem indicates
```

Comment:
This is a pattern inherited from other interfaces like `MaterialPropertyInterface`. The prefix is so that there is no name collision with however the problem is stored by the `MooseObject`, I believe